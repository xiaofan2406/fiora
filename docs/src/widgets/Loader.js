/**
 * @see https://github.com/yuanyan/halogen/blob/master/src/RingLoader.js
 */
import React from 'react';
import { css } from 'emotion';

const cssClass = css`
  display: inline-block;
  & > div {
    width: 60px;
    height: 60px;
    position: relative;
    & div {
      width: 60px;
      height: 60px;
      border-style: solid;
      border-width: 6px;
      border-color: #00bcd4;
      opacity: 0.4;
      border-radius: 100%;
      position: absolute;
      top: 0;
      left: 0;
      perspective: 800px;
      animation-fill-mode: forwards;
    }
    & div:first-child {
      animation: loaderRigthRotate 2s 0s infinite linear;
    }
    & div:nth-child(2) {
      animation: loaderLeftRotate 2s 0s infinite linear;
    }
  }
`;

const Loader = () => (
  <div className={cssClass}>
    <div>
      <div />
      <div />
    </div>
  </div>
);

export { Loader as Component };

export default Loader;
