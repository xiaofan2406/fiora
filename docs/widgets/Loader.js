/* @flow */
/**
 * @see https://github.com/yuanyan/halogen/blob/master/src/RingLoader.js
 */
import React from 'react';
import styled from 'react-emotion';

const Container = styled.div`
  display: inline-block;
  & > div {
    width: ${({ size }) => size}px;
    height: ${({ size }) => size}px;
    position: relative;
    & div {
      width: ${({ size }) => size}px;
      height: ${({ size }) => size}px;
      border-style: solid;
      border-width: ${({ size }) => size / 10}px;
      border-color: ${({ color }) => color};
      opacity: 0.4;
      border-radius: 100%;
      position: absolute;
      top: 0;
      left: 0;
      perspective: 800px;
      animation-fill-mode: forwards;
    }
    & div:first-child {
      animation: loaderRigthRotate 2s 0s infinite linear;
    }
    & div:nth-child(2) {
      animation: loaderLeftRotate 2s 0s infinite linear;
    }
  }
`;

type LoaderProps = {
  color?: string,
  size?: number,
};

const Loader = (props: LoaderProps) => (
  <Container {...props}>
    <div>
      <div />
      <div />
    </div>
  </Container>
);

Loader.defaultProps = {
  color: '#00bcd4',
  size: 60,
};

export default Loader;
