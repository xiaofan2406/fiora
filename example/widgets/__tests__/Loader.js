import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Loader from '../Loader';

it('should match snapshot', () => {
  const component = mount(<Loader />);

  expect(toJson(component)).toMatchSnapshot();
});
