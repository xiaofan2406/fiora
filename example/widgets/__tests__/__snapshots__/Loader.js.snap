// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<Loader
  color="#e8e8e8"
  size={60}
>
  <styled(Component)
    color="#e8e8e8"
    size={60}
  >
    <div
      className="css-LoaderContainer-1gt9qhw0 css-urt01t"
      color="#e8e8e8"
      size={60}
    >
      <div>
        <div />
        <div />
      </div>
    </div>
  </styled(Component)>
</Loader>
`;
